src/validation.cpp

```diff
@@ -1296,6 +1296,19 @@ void static InvalidBlockFound(CBlockIndex *pindex, const CValidationState &state
     }
 }
 
+/** Apply the effects of a block on the utxo cache, ignoring that it may already have been applied. */
+static void ReplayBlock(const CBlock& block, CCoinsViewCache& inputs, int nHeight)
+{
+    for (const CTransactionRef& tx : block.vtx) {
+        if (!tx->IsCoinBase()) {
+            for (const CTxIn &txin : tx->vin) {
+                inputs.ModifyCoins(txin.prevout.hash)->Spend(txin.prevout.n);
+            }
+        }
+        inputs.ModifyNewCoins(tx->GetHash(), tx->IsCoinBase())->FromTx(*tx, nHeight);
```

Comment:
Based on the comment above ModifyNewCoins I do not believe this works, we may need something new to capture the "maybe not fresh, but definitely fully overwrite in any case" case.